main_script.py

```diff
@@ -0,0 +1,348 @@
+
+
+def run_genscan(sequence=None, sequence_file=None, organism="Vertebrate", exon_cutoff=1, sequence_name=""):
+    """
+     Executes a GENSCAN analysis by making a request similar to filling out the form on the website.
+     Accepts all parameters that can be specified on the website (except Print options).
+
+     Args:
+     - sequence: str, optional: The sequence as a string or any convenient data type.
+     - sequence_file: str, optional: The path to the file containing the sequence, which can be uploaded and used instead of the sequence.
+     - organism: str, optional: The organism type for the analysis. Default is "Vertebrate".
+     - exon_cutoff: float, optional: The exon probability cutoff. Default is 1.00.
+     - sequence_name: str, optional: The name of the sequence. Default is an empty string.
+
+     Returns:
+     - GenscanOutput: An object containing the analysis output.
+    """
+    url = "http://hollywood.mit.edu/cgi-bin/genscanw_py.cgi"
+    data = {
+        "-o": organism,
+        "-e": exon_cutoff,
+        "-n": sequence_name,
+        "-u": sequence_file,
+        "-s": sequence,
+        "-p": "Predicted CDS and peptides"
+    }
+    if sequence:
+        data["-s"] = sequence
+    elif sequence_file:
+        with open(sequence_file, 'r') as file:
+            data["-s"] = file.read()
+    response = requests.post(url, data=data)
+    soup = BeautifulSoup(response.content, 'html.parser')
+    lines = soup.prettify().split('\n')
+    status = response.status_code
+
+    name = '---'
+    if sequence_name:
+        name = sequence_name
+    cds_list = []
+    intron_list = {}
+    exon_list = {}
+    exon_lines = [line for line in lines if re.match(r"^ \d", line)]
+    i = 1
+    end = 0
+    for l in exon_lines:
+        parts = l.strip().split()
+        exon_list[f"{parts[1]}_{i}"] = (int(parts[3]), int(parts[4]))
+        intron_list[f"{i}"] = (end + 1, int(parts[3]) - 1)
+        end = int(parts[4])
+        i += 1
+    current_seq = ""
+    no_empty_lines = [i for i in lines if i]
+    for line in no_empty_lines:
+        if line[0] in {'a', 't', 'c', 'g', 'n'}:
+            current_seq += line.strip()
+        elif current_seq != '':
+            cds_list.append(current_seq)
+            current_seq = ""
+    if current_seq:
+        cds_list.append(current_seq)
+    return GenscanOutput(status=status, name=name, cds_list=cds_list, intron_list=intron_list, exon_list=exon_list)
+
+
+class BiologicalSequence(ABC, str):
+
+    def __init__(self, seq):
+        self.seq = seq
+
+    @abstractmethod
+    def get_alphabet(self):
+        pass
+
+    def check_alphabet(self):
+        for i in self.seq:
+            if i not in self.get_alphabet():
+                raise UnexpectedSymbolInSeqError(i)
+
+
+    @abstractmethod
+    def to_print_seq(self):
+        return self.seq
+        pass
+
+
+class UnexpectedSymbolInSeqError(ValueError):

```

Comment:
Я бы наверное так вот подсократил
```suggestion
class UnexpectedSymbolError(ValueError):
```